src/libraries/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -908,6 +909,21 @@ private void BuildCookieCollectionFromDomainMatches(Uri uri, bool isSecure, int
             }
         }
 
+        // Implement path-matching according to https://tools.ietf.org/html/rfc6265#section-5.1.4
+        private static bool PathMatch(string requestPath, string cookiePath)
+        {
+            cookiePath = CookieParser.CheckQuoted(cookiePath);
+            // A request-path path-matches a given cookie-path if at least one of the following conditions holds:
+            // - The cookie-path and the request-path are identical.
+            // - The cookie-path is a prefix of the request-path, and the last character of the cookie-path is %x2F ("/").
+            // - The cookie-path is a prefix of the request-path, and the first character of the request-path that is not included in the cookie-path is a %x2F ("/") character.
+            if (!requestPath.StartsWith(cookiePath, StringComparison.Ordinal))
+                return false;
+            return requestPath.Length <= cookiePath.Length ||
```

Comment:
This check is confusing -- requestPath.Length should never be less than cookiePath.Length in this case, right? So why not just do ==?

Reply:
This could would benefit from some comments with examples.